pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1080,12 +1082,25 @@ protected void handleSubscribe(final CommandSubscribe subscribe) {
                 }
                 Optional<Map<String, String>> subscriptionProperties = SubscriptionOption.getPropertiesMap(
                         subscribe.getSubscriptionPropertiesList());
+
+                boolean createTopicIfDoesNotExist = isAuthorizedToCreateTopic && forceTopicCreation
+                        && service.isAllowAutoTopicCreation(topicName.toString());
```

Comment:
By making each of these required, it breaks the current contract, which is that permission to produce to or consume from a topic is sufficient to auto create a topic when auto topic creation is enabled.

I think it would make more sense to add this feature as an alternative to auto topic creation since we're explicitly giving a producer or consumer permission to create a topic. I think changing the PR to align with my suggestion is as simple as using `or` instead of `and` in this part of the code and any other relevant parts:

```suggestion
                boolean createTopicIfDoesNotExist = forceTopicCreation || (isAuthorizedToCreateTopic
                        && service.isAllowAutoTopicCreation(topicName.toString()));
```

Note that the `forceTopicCreation` is from the consumer, and I think we should honor that input even when the consumer has permission to create the topic.

Reply:
LGTM.